informer/disk/disk.go

```diff
@@ -12,23 +13,44 @@ import (
 	"github.com/ipfs/ipfs-cluster/api"
 )
 
+const DefaultMetric = "disk-freespace"
+
 var logger = logging.Logger("diskinfo")
 
 // MetricTTL specifies how long our reported metric is valid in seconds.
 var MetricTTL = 30
 
-// MetricName specifies the name of our metric
-var MetricName = "disk"
+// nameToRPC maps from a specified metric name to the corrresponding RPC call
+var nameToRPC = map[string]string{
+	"disk-freespace": "IPFSFreeSpace",
+	"disk-reposize":  "IPFSRepoSize",
+}
 
 // Informer is a simple object to implement the ipfscluster.Informer
 // and Component interfaces.
 type Informer struct {
-	rpcClient *rpc.Client
+	metricName string
+	rpcClient  *rpc.Client
 }
 
-// NewInformer returns an initialized Informer.
+// NewInformer returns an initialized Informer that uses the DefaultMetric
 func NewInformer() *Informer {
-	return &Informer{}
+	return &Informer{
+		metricName: DefaultMetric,
+	}
+}
+
+// NewInformerWithMetric returns an initialized Informer with a specified metric
+// name, if that name is valid, or nil. A metric name is valid if it is a key in
+// the nameToRPC map.
+func NewInformerWithMetric(metric string) *Informer {
+	// check whether specified metric is supported
+	if _, valid := nameToRPC[metric]; valid {
```

Comment:
There something here I don't fully like.

I think we need constants for allowed metric types for the disk informer:

```go
const (
  MetricFreeSpace = iota
  MetricRepoSize
)
```

`NewInformerWithMetric(metric int)` should panic and inform the user that given metric is invalid. It should also log a critical error (try to see if logger.Critical works, otherwise logger.Error will do).

`nameToRPC` becomes a map from int -> string

`Informer` can store directly the RPC name (that way there is no need to read it from the map on every GetMetric).